pom.xml

```diff
@@ -261,6 +317,19 @@
             <groupId>org.sonatype.plugins</groupId>
             <artifactId>nexus-staging-maven-plugin</artifactId>
           </plugin>
+          <plugin>
+            <groupId>com.mycila</groupId>
+            <artifactId>license-maven-plugin</artifactId>
```

Comment:
`<inherited>false</inherited>` is missing.

If what you would like to achieve is:

1. Declare and configure the plugin for parent pom projects and sub projects
2. Use the plugin in this project

The plugin must not be inherited otherwise it will get executed on all sub projects of the downstream projects. The license plugin, when used with aggregate = true is meant to be used once in the parent pom.

There is a way to define this plugin and have it executed per module in an independent way without aggregate. This requires an extra work (put the license in jar or zip, deployed in a repo) and refer it as a classpath entry with a maven dependency for the plugin). This is the cleaner way. This is how I did it in my repo of projects:

- https://github.com/mycila
- license project: https://github.com/mycila/licenses
- parent pom configuration: https://github.com/mycila/pom

Reply:
What I'm getting from all of this is:
* It is impossible to fully encapsulate license plugin in the parent pom without it running on all child project's submodules.  The child project will have to redefine at least some part of it.
* It may be easier not to try this at all
* It would work fine (as it is) to permit the license plugin to run on each of the child project's submodules.  In my testing, this takes milliseconds.  I know that you are not comfortable with this.

My preference is to go with the last option (current state of the PR).  If that's not acceptable, I'll abandon doing anything with the license project.

Reply:
Did you look at the examples I gave you ? 
The license plugin works in different ways. It can be configured to work like you want. It just needs to be configured correctly.

- it can work globally on the project (agrgegate=true, inherited=false), executed at parent level only
- it can be executed with each module (agrgegate=false inherited=true)

Currently at Terracotta we were lazy and we were copying it over projects because we didn't have any parent pom in place.

So we can either configure the plugin correctly so that it can go in the parent pom, or leave it as-is in the sub projects ;-)